WordPress/Classes/Services/MediaService.m

```diff
@@ -16,140 +16,8 @@
 
 NSErrorDomain const MediaServiceErrorDomain = @"MediaServiceErrorDomain";
 
-@interface MediaService ()
-
-@property (nonatomic, strong) MediaThumbnailService *thumbnailService;
-
-@end
-
 @implementation MediaService
 
-- (instancetype)initWithManagedObjectContext:(NSManagedObjectContext *)context
-{
-    self = [super initWithManagedObjectContext:context];
-    if (self) {
-        _concurrentThumbnailGeneration = NO;
-    }
-    return self;
-}
-
-#pragma mark - Creating media
-
-- (Media *)createMediaWith:(id<ExportableAsset>)exportable
-                      blog:(Blog *)blog
-                      post:(AbstractPost *)post
-                  progress:(NSProgress **)progress
-         thumbnailCallback:(void (^)(Media *media, NSURL *thumbnailURL))thumbnailCallback
-                completion:(void (^)(Media *media, NSError *error))completion
-{
-    NSParameterAssert(post == nil || blog == post.blog);
-    NSParameterAssert(blog.managedObjectContext == self.managedObjectContext);
-    NSProgress *createProgress = [NSProgress discreteProgressWithTotalUnitCount:1];
-    __block Media *media;
-    __block NSSet<NSString *> *allowedFileTypes = [NSSet set];
-    [self.managedObjectContext performBlockAndWait:^{
-        if ( blog == nil ) {
-            if (completion) {
-                NSError *error = [NSError errorWithDomain: MediaServiceErrorDomain code: MediaServiceErrorUnableToCreateMedia userInfo: nil];
-                completion(nil, error);
-            }
-            return;
-        }
-        
-        if (blog.allowedFileTypes != nil) {
-            // HEIC isn't supported when uploading an image, so we filter it out (http://git.io/JJAae)
-            NSMutableSet *mutableAllowedFileTypes = [blog.allowedFileTypes mutableCopy];
-            [mutableAllowedFileTypes removeObject:@"heic"];
-            allowedFileTypes = mutableAllowedFileTypes;
-        }
-
-        if (post != nil) {
-            media = [Media makeMediaWithPost:post];
-        } else {
-            media = [Media makeMediaWithBlog:blog];
-        }
-        media.mediaType = exportable.assetMediaType;
-        media.remoteStatus = MediaRemoteStatusProcessing;
-
-        [self.managedObjectContext obtainPermanentIDsForObjects:@[media] error:nil];
-        [[ContextManager sharedInstance] saveContextAndWait:self.managedObjectContext];
-    }];
-    if (media == nil) {
-        return nil;
-    }
-    NSManagedObjectID *mediaObjectID = media.objectID;
-    [self.managedObjectContext performBlock:^{
-        // Setup completion handlers
-        void(^completionWithMedia)(Media *) = ^(Media *media) {
-            media.remoteStatus = MediaRemoteStatusLocal;
```

Comment:
⚠️ Potential issue: `media` is modified in the main thread since this `completionWithMedia` is [called from the main thread](https://github.com/wordpress-mobile/WordPress-iOS/pull/20322/files#r1136467532), but there is no guarantee `media` is bound to the main context.